### PR TITLE
Add script for upgrade preflight check

### PR DESCRIPTION
**Pull request by bk201**
Add pre-upgrade check script for v1.1.3.

### CODE REVIEW

File: pre-check/v1.0.3/check.sh [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+}
+
+check_volumes()
+{
+    echo ">>> Check Longhorn volumes..."
+
+    volumes=$(kubectl get volumes.longhorn.io -A -o yaml)
+
+    # all volumes should be healthy
+    degraded=$(echo "$volumes" | yq '.items | any_c(.status.state == "attached" and .status.robustness != "healthy")')
```
Comment (starbops) [inline pre-check/v1.0.3/check.sh:141]: Do we want to skip the check when running the script against a single-node cluster?

**Review by starbops — COMMENTED** [review]
Mostly LGTM, just one nit.

File: pre-check/v1.0.3/check.sh (hunk 1 above)
Comment (bk201) [reply pre-check/v1.0.3/check.sh]: Good suggestion, I added checks in https://github.com/harvester/upgrade-helpers/pull/3/commits/b074f0226fa296e954f8f064d327b97bf6d11a68
And I also fix the problem that inside a piped while scope, we can't set the global variable.

**Review by starbops — APPROVED** [review]
LGTM, thanks!

File: pre-check/v1.0.3/check.sh [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    if [ "$cluster_phase" != "Provisioned" ]; then
+        echo "Cluster is not provisioned ($cluster_phase)"
+        record_fail
+        return
+    fi
+
+    echo "The CAPI cluster is provisioned."
+}
+
+check_machines()
```
Comment (guangbochen) [inline pre-check/v1.0.3/check.sh:117]: Might be good to add node count checks? to make sure that there are no orphan machines.

**Review by guangbochen — COMMENTED** [review]
LGTM and verified that all checks are

File: pre-check/v1.0.3/check.sh (hunk 2 above)
Comment (bk201) [reply pre-check/v1.0.3/check.sh:117]: Added! https://github.com/harvester/upgrade-helpers/pull/3/commits/545724525ce7f91c1a3aaf62440ea5f05a94811d

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**